services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -5680,7 +5770,16 @@ public String getNativeReplicationSourceFabric(
     }
 
     if (sourceFabric == null || sourceFabric.isEmpty()) {
-      sourceFabric = multiClusterConfigs.getControllerConfig(clusterName).getNativeReplicationSourceFabric();
+      sourceFabric = getMultiClusterConfigs().getControllerConfig(clusterName).getNativeReplicationSourceFabric();
+    }
+
+    if (targetedRegions != null) {
+      Set<String> regions = RegionUtils.parseRegionsFilterList(targetedRegions);
+      if (!regions.contains(sourceFabric) && !emergencySourceRegion.isPresent()) {
```

Comment:
Add a detailed log here indicating the target colo list, NR fabric and which one we are choosing.

Reply:
done